source/common/tls/cert_validator/san_matcher.cc

```diff
@@ -44,6 +49,15 @@ SanMatcherPtr createStringSanMatcher(
     return SanMatcherPtr{std::make_unique<StringSanMatcher>(GEN_URI, matcher.matcher(), context)};
   case envoy::extensions::transport_sockets::tls::v3::SubjectAltNameMatcher::IP_ADDRESS:
     return SanMatcherPtr{std::make_unique<StringSanMatcher>(GEN_IPADD, matcher.matcher(), context)};
+  case envoy::extensions::transport_sockets::tls::v3::SubjectAltNameMatcher::OTHER_NAME: {
+    // Invalid/Empty OID returns a nullptr from OBJ_txt2obj
+    bssl::UniquePtr<ASN1_OBJECT> oid(OBJ_txt2obj(matcher.oid().c_str(), 0));
+    if (oid == nullptr) {
+      return nullptr;
+    };
+    return SanMatcherPtr{std::make_unique<StringSanMatcher>(GEN_OTHERNAME, matcher.matcher(),
+                                                            context, std::move(oid))};
```

Comment:
just write like this:
` return std::make_unique<StringSanMatcher>(GEN_OTHERNAME, matcher.matcher(),
                                                            context, std::move(oid));`, it should be enough.

Reply:
The return values for all other cases(L44-51) are also structured like that, which is why I followed it for consistency. Do you want me to change it for all?

Reply:
And all the testcases. Might be better addressed as a separate PR?

Reply:
ok, it's fine with me